apps/application-system/api/src/app/modules/application/tools/application.serializer.ts

```diff
@@ -3,42 +3,58 @@ import {
   NestInterceptor,
   ExecutionContext,
   CallHandler,
+  Inject,
 } from '@nestjs/common'
+import { classToPlain, plainToClass } from 'class-transformer'
 import { Observable } from 'rxjs'
 import { map } from 'rxjs/operators'
 import { Application } from '../application.model'
 import {
   Application as BaseApplication,
   ApplicationTemplateHelper,
   ApplicationTypes,
+  ApplicationStateMeta,
 } from '@island.is/application/core'
 import { getApplicationTemplateByTypeId } from '@island.is/application/template-loader'
+import { ApplicationResponseDto } from '../dto/application.response.dto'
+
+const role = 'applicant' // TODO get real role
 
 @Injectable()
 export class ApplicationSerializer
-  implements NestInterceptor<Application, Promise<BaseApplication>> {
+  implements NestInterceptor<Application, Promise<any>> {
   intercept(
     context: ExecutionContext,
     next: CallHandler,
-  ): Observable<Promise<BaseApplication>> {
+  ): Observable<Promise<any>> {
     // code here will be executed before the controller executes
     return next.handle().pipe(
-      map(async (applicationModel: Application) => {
-        const application = applicationModel.toJSON() as BaseApplication
-        const template = await getApplicationTemplateByTypeId(
-          application.typeId as ApplicationTypes,
-        )
+      map(async (res: Application | Array<Application>) => {
+        const isArray = Array.isArray(res)
 
-        return {
-          ...application,
-          ...new ApplicationTemplateHelper(
-            application,
-            template,
-          ).getPermittedAnswersAndExternalData(
-            application.state === 'inReview' ? 'reviewer' : 'applicant',
-          ), // TODO how do we know the role
-        }
+        return isArray
+          ? Promise.all(
+              (res as Application[]).map((item) => this.serialize(item)),
+            )
+          : this.serialize(res as Application)
       }),
     )
   }
+
+  async serialize(model: Application) {
+    const application = model.toJSON() as BaseApplication
+    const template = await getApplicationTemplateByTypeId(
+      application.typeId as ApplicationTypes,
+    )
+    const helper = new ApplicationTemplateHelper(application, template)
+
+    const dto = plainToClass(ApplicationResponseDto, {
+      ...application,
+      ...helper.getPermittedAnswersAndExternalData(role),
```

Comment:
you can do `...helper.getPermittedAnswersAndExternalData(template.mapUserToRole([nationalRegistryId])`

but I don't know how to get the `nationalRegistryId` at this specific place